chainer/iterators/dali_iterator.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import division
+
+from chainer.dataset import iterator
+
+
+class DaliIterator(iterator.Iterator):
+
+    """Iterator for DALI pipeline.
+
+    Args:
+        pipeline: DALI pipeline.
+        repeat (bool): If ``True``, it infinitely loops over the dataset.
+            Otherwise, it stops iteration at the end of the first epoch.
+
+    """
+
+    def __init__(self, pipeline, repeat=True):
+        pipeline.build()
```

Comment:
Does this operation cause CUDA context initialization? In that case I think it's better to move it to `__next__` to defer initialization.

Reply:
Though I'm not sure if CUDA context is created there, I agree with you that it's safer to move it to `__next__`.